### PR TITLE
[Impeller] Render emoji shadows with correct colors.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/138819

Emoji glyphs normally ignore their text color (except for alpha channel). Text shadows are created by rendering the font with the shadow color as the text color and then applying a blur. If this is done to an emoji, we need to apply the text color so that the shadow has the correct color.

### CODE REVIEW

File: impeller/entity/contents/text_contents.h [hunk 1]
```diff
@@ -30,6 +30,10 @@ class TextContents final : public Contents {
 
   void SetColor(Color color);
 
+  /// @brief Force the text color to apply to the rendered glyphs, even if those
+  ///        glyphs are bitmaps.
```
Comment (bdero) [inline impeller/entity/contents/text_contents.h:34]: nit: add note that this is a trick to get color glyphs behaving correctly for the mask blur case?

**Review by bdero — APPROVED** [review]
LGTM!

File: impeller/entity/shaders/glyph_atlas_color.frag [hunk 2]
```diff
@@ -16,5 +21,9 @@ out f16vec4 frag_color;
 
 void main() {
   f16vec4 value = texture(glyph_atlas_sampler, v_uv);
-  frag_color = value * v_text_color.aaaa;
+  if (frag_info.use_text_color == 1.0f) {
```
Comment (chinmaygarde) [inline impeller/entity/shaders/glyph_atlas_color.frag:24]: Nit: The floating point comparison makes me nervous. Perhaps `frag_color = mix(value * v_text_color.aaaa, value.aaaa * v_text_color, frag_info.use_text_color)` instead?

Comment (jonahwilliams) [reply impeller/entity/shaders/glyph_atlas_color.frag:24]: Why does it make you nervous?

File: impeller/entity/contents/text_contents.h (hunk 1 above)
Comment (jonahwilliams) [reply impeller/entity/contents/text_contents.h:34]: Done

File: impeller/entity/shaders/glyph_atlas_color.frag (hunk 2 above)
Comment (jonahwilliams) [reply impeller/entity/shaders/glyph_atlas_color.frag:24]: I'm going to merge this now. Please file a bug describing the problem you'd like me to address and I'll handle it later.

**Pull request merged**